### DISCUSSION TITLE
Test updating eModbus library to version 1.6.0

### DISCUSSION THREAD

**Discussion by dingo35** [serkri/SmartEVSE-3#125] [Announcements]
[allin-eModbus-1.6.0.zip](https://github.com/serkri/SmartEVSE-3/files/11200434/allin-eModbus-1.6.0.zip)

The original eModbus library is 1.4.1, which Stegen has patched for unstable modbus installations. It is unclear if version 1.6.0 still needs the patch. 

Please test and post your results here!

**Comment by rvdgaag** [comment]
Just uploaded the (SPIFFS) file, the smartevse is not responding anymore. How can I get it back to work ?

**Comment by dingo35** [comment]
So the normal procedure is, first flash firmware.bin then spiffs.bin .

In case of problems: http://youripaddress/update should still give you the flash screen, so you can resume flashing...

EDIT: In my setup no modbus or webserver problems....

**Reply by rvdgaag** [reply]
Ok it was not responding on the wifi, but now it is again and I installed the FW. Charging the car now

**Reply by rvdgaag** [reply]
Seems to be running smooth also on the modbus section

**Comment by dingo35** [comment]
Ok did you have problems on your modbus on older versions? How long is your modbus cable? Is the modbus terminated by resistors or did you leave it open?

**Reply by rvdgaag** [reply]
I have lot of CRC issues, still. Cable is 10 meters long and terminated. Will replace the cable probably next week with a special modbus cable I bought, replacing the CAT5 I'm using now.

**Comment by dingo35** [comment]
[emodbus-1.6.0-patched.zip](https://github.com/serkri/SmartEVSE-3/files/11202733/emodbus-1.6.0-patched.zip)

Here is the patched version, does this reduce the number of CRC errors?

**Reply by rvdgaag** [reply]
No, same

**Comment by dingo35** [comment]
Too bad!  There is a reworked version 1.7.0 that needs adaptions; perhaps that improves it, Ill try to adapt the code tomorrow.

**Comment by dingo35** [comment]
[1.7.0-unpatched.zip](https://github.com/serkri/SmartEVSE-3/files/11208220/1.7.0-unpatched.zip)

Hope this brings any relief in your CRC errors .... no change in spiffs.bin so no need to flash that.

**Reply by rvdgaag** [reply]
Thanks, no difference yet, lets see what the new cabling will bring.

**Comment by dingo35** [comment]
[1.7.0-patched.zip](https://github.com/serkri/SmartEVSE-3/files/11210258/1.7.0-patched.zip)
Ok I am starting to think this is not a software problem, but if you would test this last version....

It is very strange, my modbus cable is 20-25m long, consists of different kinds of cabling connected together (old ISDN cable w/ twisted pair, untwisted "bel-kabel", and _no_ errors on the line.
How are you detecting your CRC errors? Are they all from the same device? Perhaps one of the devices is (partly) broken?

**Reply by fluppie** [reply]
On long cables I also had this, connected GND from the Eastron modbus meters and my problems were history.
